File: native/ios/Sources/Core/LynxNotificationsEventForwarder.swift
import Foundation

public final class LynxNotificationsEventForwarder {
  private let module: LynxNotificationsModule

  public init(module: LynxNotificationsModule) {
    self.module = module
  }

  public func onForegroundNotificationReceived(notification: [String: Any]) {
    LynxNotificationsLogger.debug("Forwarding foreground notification event to Lynx module.")
    module.emitNotificationReceived(notification: notification)
  }

  public func onNotificationResponse(response: [String: Any]) {
    LynxNotificationsLogger.debug("Forwarding notification response event to Lynx module.")
    module.emitNotificationResponse(response: response)
  }

  public func onTokenRefreshed(token: String) {
    LynxNotificationsLogger.debug("Forwarding token refresh event to Lynx module.")
    module.emitTokenRefreshed(token: PushToken(type: "fcm", data: token))
  }
}

File: native/android/core/src/main/java/io/lynx/notifications/core/LynxNotificationsEventForwarder.java
package io.lynx.notifications.core;

import java.util.Map;

/**
 * Adapts host push lifecycle callbacks to Lynx notifications module events.
 */
public final class LynxNotificationsEventForwarder {
  private final LynxNotificationsModule module;

  public LynxNotificationsEventForwarder(LynxNotificationsModule module) {
    this.module = module;
  }

  public void onForegroundNotificationReceived(Map<String, Object> notification) {
    LynxNotificationsLogger.debug("Forwarding foreground notification event to Lynx module.");
    module.emitNotificationReceived(notification);
  }

  public void onNotificationResponse(Map<String, Object> response) {
    LynxNotificationsLogger.debug("Forwarding notification response event to Lynx module.");
    module.emitNotificationResponse(response);
  }

  public void onTokenRefreshed(String token) {
    LynxNotificationsLogger.debug("Forwarding token refresh event to Lynx module.");
    module.emitTokenRefreshed(new PushToken("fcm", token));
  }
}
